src/plugins/discover/public/application/components/discover.tsx

```diff
@@ -119,150 +169,161 @@ export function Discover({
           showSearchBar={true}
           useDefaultBehaviors={true}
         />
+        {flyout}
         <main className="dscApp__frame">
-          <EuiResizableContainer>
-            {(EuiResizablePanel, EuiResizableButton) => (
-              <>
-                <EuiResizablePanel
-                  minSize="20%"
-                  initialSize={100}
-                  scrollable={false}
-                  className="dscApp__sidebar"
-                >
-                  <DiscoverSidebar
-                    columns={state.columns}
-                    fieldCounts={fieldCounts}
-                    hits={rows}
-                    indexPatternList={indexPatternList}
-                    onAddField={addColumn}
-                    onAddFilter={onAddFilter}
-                    onRemoveField={onRemoveColumn}
-                    selectedIndexPattern={searchSource && searchSource.getField('index')}
-                    setIndexPattern={setIndexPattern}
-                  />
-                </EuiResizablePanel>
+          <>
+            <div className="dscSidebar dscSidebar__desktop hidden-xs hidden-sm">
+              <DiscoverSidebar
+                columns={state.columns}
+                fieldCounts={fieldCounts}
+                hits={rows}
+                indexPatternList={indexPatternList}
+                onAddField={addColumn}
+                onAddFilter={onAddFilter}
+                onRemoveField={onRemoveColumn}
+                selectedIndexPattern={searchSource && searchSource.getField('index')}
+                setIndexPattern={setIndexPattern}
+              />
+            </div>
+            <div className="dscSidebar dscSidebar__mobile visible-xs visible-sm">
+              <DiscoverSidebarMobile
+                columns={state.columns}
+                fieldCounts={fieldCounts}
+                hits={rows}
+                indexPatternList={indexPatternList}
+                onAddField={addColumn}
+                onAddFilter={onAddFilter}
+                onRemoveField={onRemoveColumn}
+                selectedIndexPattern={searchSource && searchSource.getField('index')}
+                setIndexPattern={setIndexPattern}
+              />
+              <EuiButton
+                contentProps={{ className: 'dscSidebar__mobileButton' }}
+                iconSide="right"
+                iconType="arrowRight"
+                fullWidth
+                onClick={() => setIsFlyoutVisible(true)}
+              >
+                <FormattedMessage
+                  id="discover.fieldChooser.fieldsMobileButtonLabel"
+                  defaultMessage="Fields"
+                />
+                <EuiBadge className="dscSidebar__mobileBadge" color="accent">
+                  5
```

Comment:
to start this: the columns are an array `state.columns` so you can display it's length. however, by default `_source` is selected, and this would count as 1. no sure if this would be better as `0`... however we can decide later

Reply:
So I guess we would go with 1 in that case then as it seems like there's never zero **Selected Fields**.

Reply:
Yes, let's handle it that way 👍

Reply:
@kertal so I discussed this with Caroline and we'd prefer not to count `_source `as a field. The logic is that `_source` is not really a field and can be misleading for users to place it under `Selected Fields`. What we're proposing is to only show the `Selected fields` accordion once the user has chosen to add at least one field.

<img width="700" alt="image 46" src="https://user-images.githubusercontent.com/4016496/94622853-c54d5a00-0267-11eb-9e1d-4d5fc0c7ffcd.png">

Reply:
yes, it's just the default, we should not count it